From fa0d1c9afd7a352b08fee32b67b210848e499831 Mon Sep 17 00:00:00 2001
From: Ferry Toth <ftoth@exalondelft.nl>
Date: Sat, 21 Jan 2023 15:13:26 +0100
Subject: [PATCH 1/1] jobserver create queue in TMPDIR

Certains hosts may be running multiple bitbake's in parallel. Each
should have their own job queue.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>
---
 bitbake/lib/bb/runqueue.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bitbake/lib/bb/runqueue.py b/bitbake/lib/bb/runqueue.py
index 072f2d49b7..57fbbd3afd 100644
--- a/bitbake/lib/bb/runqueue.py
+++ b/bitbake/lib/bb/runqueue.py
@@ -1478,7 +1478,7 @@ class RunQueue:
         return bb.utils.better_eval(call, locs)
 
     def setup_make_fifo(self):
-        fifoname = "/tmp/makefifo"
+        fifoname = self.cfgData.getVar("TMPDIR") + "/makefifo"
         self.cfgData.setVar("BB_MAKEFIFO", fifoname)
         m = re.search(r'-j (\d+)', self.cfgData.getVar("PARALLEL_MAKE"))
         if m:
-- 
2.37.2

